### PR TITLE
[SPARK-47271][DOCS] Explain importance of statistics on SQL performance tuning page

### PR DESCRIPTION
**Pull request by nchammas**
### What changes were proposed in this pull request?

Add a new section to the SQL performance tuning page on statistics.

Tweak some related documentation relevant to this new section.

### Why are the changes needed?

Statistics are a basic and critical part of how Spark optimizes queries and achieves high performance. Yet, they receive no discussion in our performance tuning guide.

### Does this PR introduce _any_ user-facing change?

Yes, new user-facing documentation.

### How was this patch tested?

Built and reviewed the docs locally.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: docs/sql-performance-tuning.md [hunk 1]
```diff
@@ -157,6 +157,18 @@ SELECT /*+ REBALANCE(3, c) */ * FROM t;
 
 For more details please refer to the documentation of [Partitioning Hints](sql-ref-syntax-qry-select-hints.html#partitioning-hints).
 
+## Leveraging Statistics
+Apache Spark's ability to choose the best execution plan among many possible options is determined in part by its estimates of how many rows will be output by every node in the execution plan (read, filter, join, etc.). Those estimates in turn are based on statistics that are made available to Spark in one of several ways:
+
+- **Data source**: Statistics that Spark reads directly from the underlying data source, like the counts and min/max values in the metadata of Parquet files. These statistics are maintained by the underlying data source.
+- **Catalog**: Statistics that Spark reads from the catalog, like the Hive Metastore. These statistics are collected or updated whenever you run [`ANALYZE TABLE`](sql-ref-syntax-aux-analyze-table.html).
+- **Runtime**: Statistics that Spark computes itself at runtime as a job is running.
```
Comment (xinrong-meng) [inline docs/sql-performance-tuning.md:165]: I'm curious what statistics are considered runtime? Do we want to add examples here?

**Review by xinrong-meng — APPROVED** [review]

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (nchammas) [reply docs/sql-performance-tuning.md:165]: I believe this concept is specific to adaptive query execution. It was [introduced here][1]. I'll clarify this point accordingly.

[1]: https://github.com/apache/spark/pull/32391/files#diff-c42595588570b41aeb8e33e98553522a53cd46751abfc98b1af3495cf4f5bc15R50-R51

**Review by GideonPotok — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -582,11 +582,7 @@ object SQLConf {
 
   val AUTO_BROADCASTJOIN_THRESHOLD = buildConf("spark.sql.autoBroadcastJoinThreshold")
     .doc("Configures the maximum size in bytes for a table that will be broadcast to all worker " +
-      "nodes when performing a join.  By setting this value to -1 broadcasting can be disabled. " +
-      "Note that currently statistics are only supported for Hive Metastore tables where the " +
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:586]: qq: why do we remove this?

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:586]: The comment `are only supported for Hive Metastore tables` seems inaccurate.

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (nchammas) [reply docs/sql-performance-tuning.md]: I've explained this a bit more here and a few lines below. I've also updated the screenshot in the PR description.

Comment (xinrong-meng) [reply docs/sql-performance-tuning.md]: Thank you @nchammas , that's very helpful!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 2 above)
Comment (nchammas) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:586]: Yes, this comment was added around 9 years ago and is out of date.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**